backends/bmv2/common/deparser.cpp

```diff
@@ -69,7 +78,8 @@ Util::IJson* DeparserConverter::convertDeparser(const IR::P4Control* ctrl) {
     result->emplace("id", nextId("deparser"));
     result->emplace_non_null("source_info", ctrl->sourceInfoJsonObj());
     auto order = mkArrayField(result, "order");
-    convertDeparserBody(&ctrl->body->components, order);
+    auto primitives = mkArrayField(result, "primitives");
```

Comment:
how come we didn't need this array previously?

Reply:
Because there is no primitives in deparser. I needed some array to store log_msg primitives.
There are some limitations of this solution such as:

log_msg("A")
emit(..)
log_msg("B")

and 
log_msg("A")
log_msg("B")
emit(..)

have same outputs because this solution doesn't store emit and log_msg in same array.
If this is a problem, my suggestion is to store them together in one array and distinguish them by some sort of type.